camel/functions/web_search.py

```diff
@@ -0,0 +1,243 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import os
+from typing import Any, Dict, List
+
+import requests
+from bs4 import BeautifulSoup
+
+import camel.agents
+from camel.messages import BaseMessage
+
+from .openai_function import OpenAIFunction
+
+
+def search_google(query: str) -> List[Dict[str, Any]]:
+    r"""using google search engine to search information for the given query.
+
+    Args:
+        query (string): what question to search.
+
+    Returns:
+        List: a list of web information, include title, descrption, url.
+    """
+    # https://developers.google.com/custom-search/v1/overview
+    GOOGLE_API_KEY = os.getenv("GOOGLE_API_KEY")
+    # https://cse.google.com/cse/all
+    SEARCH_ENGINE_ID = os.getenv("SEARCH_ENGINE_ID")
+
+    # using the first page
+    start = 1
+    # different language may get different result
+    language = "en"
+    # how many pages to return
+    numbers = 10
+    # constructing the URL
+    # doc: https://developers.google.com/custom-search/v1/using_rest
+    url = f"https://www.googleapis.com/customsearch/v1?" \
+          f"key={GOOGLE_API_KEY}&cx={SEARCH_ENGINE_ID}&q={query}&start=" \
+          f"{start}&lr={language}&num={numbers}"
+
+    responses = []
+    # make the get
```

Comment:
Misplaced comment I guess. Put it just above the `requests.get(url)` and use another clearer expression like "Fetch the results given the URL"